documentation/directory-structure.md

```diff
@@ -0,0 +1,247 @@
+  - [`stable` and `preview` folders](#stable-and-preview-folders)
+  - [A complete example directory structure of `specification/<azureTeam>`](#a-complete-example-directory-structure-of-specificationazureteam)
+  - [Naming guidelines for `specification` folder contents](#naming-guidelines-for-specification-folder-contents)
+  - [`specification/common-types`](#specificationcommon-types)
+  - [Advanced scenario: service group](#advanced-scenario-service-group)
+    - [Service group `common-types`](#service-group-common-types)
+    - [Versioning services in a service group](#versioning-services-in-a-service-group)
+  - [Deprecated structure and hand-written OpenAPI specs](#deprecated-structure-and-hand-written-openapi-specs)
+# `specification` directory structure
+
+
+This article describes the directory structure / folder layout of the `specification` folder.
+You may be also interested in following:
+- [Specification index]
+- [Resource Provider list]
+
+> [!IMPORTANT]
+> The structure described in this article is strictly enforced. There exist some exceptions for historical reasons.
+> These exceptions are not allowed going forward.
+
+## `specification` folder
+
+The `specification` folder is the root folder for all service specifications.
+
+Each child of the `specification` folder corresponds to a `service` specification for given Azure team. Here we denote such folder as `<azureTeam>`.
+In advanced cases for big teams the `<azureTeam>` folder can host multiple services, known as `service group`.
+Read [the relevant section](#advanced-scenario-service-group) for details.
+
+Given `<azureTeam>` has following structure:
+
+- `<azureTeam>/<typeSpecFolder>` (multiple folders)
+- `<azureTeam>/resource-manager`
+- `<azureTeam>/data-plane`
+
+The `<azureTeam>/<typeSpecFolder>` folders contain the TypeSpec specification for the given `service` or `service group`.
+You can find details on the name and contents of these folders in [TypeSpec directory structure].
+You can learn more about TypeSpec at [aka.ms/azsdk/typespec] and [aka.ms/typespec].
+
+## `resource-manager` and `data-plane` folders
+
+The `<azureTeam>/resource-manager` contains the ARM OpenAPI specifications emitted from TypeSpec in `<azureTeam>/<typeSpecFolder>`.
+
+The `resource-manager` folder has exactly one child folder whose name matches the **Resource Provider (RP) Namespace** (`<RPNS>`), 
+such as `Microsoft.Automation` (full list of namespaces is [here][Resource Provider list]).
+There is 1 to 1 correspondence between an RP and an RP namespace.
+There must be **exactly one** RP namespace folder under given `resource-manager` folder.
+We denote such folder as `resource-manager/<RPNS>`.
+
+The `<azureTeam>/data-plane` contains the data-plane OpenAPI specifications emitted from TypeSpec in `<typeSpecFolder>`. 
+The `data-plane` folder has no child `<RPNS>` folder. However, it can have a set of `<dataPlaneSubfolder>` folders.
+
+## AutoRest configuration `README.md` files
```

Comment:
The README.md is used for more that autorest so we should not call this an "autorest configuration" file.

Reply:
Ungh. We call them like that, like, _everywhere_. I think it is fine to keep it as-is. These README files are frankenstein creations anyway. Plus it is _technically_ true they are AutoRest config files, the best kind of true ;)